storagemarket/deal_execution.go

```diff
@@ -596,6 +596,67 @@ func (p *Provider) indexAndAnnounce(ctx context.Context, pub event.Emitter, deal
 	return p.updateCheckpoint(pub, deal, dealcheckpoints.IndexedAndAnnounced)
 }
 
+// fireSealingUpdateEvents periodically checks the sealing status of the deal
+// and fires events for each change
+func (p *Provider) fireSealingUpdateEvents(pub event.Emitter, dealUuid uuid.UUID, sectorNum abi.SectorNumber) {
+	var lastSealingState lapi.SectorState
```

Comment:
They should see it immediately - when they retrieve the deal it will already show the sealing state.
Also when fireSealingUpdateEvents is called it immediately checks the status (see line 621)

Reply:
But shouldn't we see an event for that sealing state update to `Proving` ?

Reply:
Oh nvm, I see `checkstatus` also publishes the event.